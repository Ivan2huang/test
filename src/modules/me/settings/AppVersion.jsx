import React from 'react';
import { injectIntl } from 'react-intl';
import * as PropTypes from 'prop-types';

import { Box } from '@material-ui/core';

import { Typography } from '../../../uiComponents';
import CONFIG from '../../../constants/config';
import { formatMessage } from '../../../helpers/helpers';

const AppVersion = ({ intl }) => {
  const appVersion = formatMessage(
    intl,
    'me.tab.setting.label.appVersion',
    `Version ${CONFIG.appVersionName}`,
    {
      appVersion: CONFIG.appVersionName,
    },
  );
  const displayEnvironment =
    CONFIG.environment && CONFIG.environment.toUpperCase() !== 'PRODUCTION';

  return (
    <Box mt={4} mb={{ xs: 4, md: 0 }} textAlign={{ xs: 'center', md: 'right' }}>
      <Typography type="style-6">
        {displayEnvironment
          ? `${appVersion} [${CONFIG.environment.toUpperCase()}]`
          : appVersion}
      </Typography>
    </Box>
  );
};

AppVersion.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default injectIntl(AppVersion);
