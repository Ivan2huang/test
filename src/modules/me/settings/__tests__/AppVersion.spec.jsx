import React from 'react';
import { render } from '@testing-library/react';

import withIntl from '../../../../i18n/withIntlProvider';
import AppVersion from '../AppVersion';
import CONFIG from '../../../../constants/config';

jest.mock('../../../../constants/config', () => ({
  appVersionName: '1.0.0',
  environment: 'uat',
  defaultLanguage: 'en',
}));

jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, message) => message),
  isEmpty: jest.fn(
    obj => Object.entries(obj).length === 0 && obj.constructor === Object,
  ),
}));

describe('App version', () => {
  const setUp = () => {
    const Component = withIntl(AppVersion);
    return render(<Component />);
  };
  it('should match snapshot for UAT environment', () => {
    const { container } = setUp();

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for no environment variable', () => {
    CONFIG.environment = '';
    const { container } = setUp();

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for PRODUCTION environment', () => {
    CONFIG.environment = 'PRODUCTION';
    const { container } = setUp();

    expect(container).toMatchSnapshot();
  });
});
