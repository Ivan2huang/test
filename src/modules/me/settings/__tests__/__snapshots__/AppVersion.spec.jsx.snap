// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App version should match snapshot for PRODUCTION environment 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-70"
  >
    <div
      class="MuiTypography-root makeStyles-root-71 makeStyles-root-72 MuiTypography-body1"
    >
      Version 1.0.0
    </div>
  </div>
</div>
`;

exports[`App version should match snapshot for UAT environment 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiTypography-root makeStyles-root-3 makeStyles-root-4 MuiTypography-body1"
    >
      Version 1.0.0 [UAT]
    </div>
  </div>
</div>
`;

exports[`App version should match snapshot for no environment variable 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-36"
  >
    <div
      class="MuiTypography-root makeStyles-root-37 makeStyles-root-38 MuiTypography-body1"
    >
      Version 1.0.0
    </div>
  </div>
</div>
`;
